databricks/koalas/series.py

```diff
@@ -657,10 +657,27 @@ def eq(self, other) -> bool:
         d    False
         Name: b, dtype: bool
         """
-        return self == other
+        if isinstance(other, (list, tuple)):
+            other = ks.Index(other, name=self.name)
+        # pandas always returns False for all items with dict and set.
+        elif isinstance(other, (dict, set)):
+            return self != self
+        return IndexOpsMixin.__eq__(self, other)
 
     equals = eq
 
+    def __eq__(self, other):
+        if isinstance(other, (list, tuple)):
+            other = ks.Index(other, name=self.name)
+        # pandas always returns False for all items with dict and set.
+        elif isinstance(other, (dict, set)):
+            return self != self
+        # pandas doesn't support `==` for different Index, but support for `eq` function.
+        elif isinstance(other, ks.Series):
+            if not self.index.equals(other.index):
+                raise ValueError("Can only compare identically-labeled Series objects")
+        return IndexOpsMixin.__eq__(self, other)
+
```

Comment:
we cannot use same code for both `eq` and `__eq__` because the behavior is different in pandas.